docs/Developer Documentation/Logger/index.md

```diff
@@ -0,0 +1,159 @@
+
+## Logger singleton with plugins 
+
+SDL plugins are shared libraries, so `LoggerSingleton` could not be implemented with Mayers singleton. 
+Mayers singleton would create own SDL logger instance for each plugin.
+
+The idea is to pass singleton pointer to Plugin during creation, so that plugin could initialize `Logger::instance` pointer with one received from SDL core. 
+
+
+Singleton Instance implementation : 
+```cpp
+// ilogger.h
+static Logger& instance(Logger* pre_init = nullptr);
+
+...
+// logger_impl.cc
+Logger& Logger::instance(Logger* pre_init) {
+  static Logger* instance_ = nullptr;
+  if (pre_init) {
+    assert(instance_ == nullptr);
+    instance_ = pre_init;
+  }
+  assert(instance_);
+  return *instance_;
+}
+```
+
+`pre_init` is `nullptr` by default, so all components will access instance_ static pointer for logging. 
+`main()`  function need to create `SDLLogger` implementation and call `Logger::instance` (logger implementation);
+
+Plugin implementation:
+```cpp 
+extern "C" PluginType* Create(Logger* logger_singleton_instance) {
+  Logger::instance(logger_instance);
+  return new PluginType();
+}
+```
+
+SDL Core part will pass pointer to logger singleton to the plugin so that plugin shared lib could initialize `Logger::instance` with the same pointer as core part. 
+
+## Logger detailed design :
+
+Each source file creates `logger_` variable via macro `SDL_CREATE_LOG_VARIABLE`. 
+This variable is actually a string with component name of the logger.
+Some logger implementations (like log4cxx) may have separate severity or destination rules for each component. 
+
+
+SDL implements all info required for log message :
+
+ * LogLevel enum
+ * Location info struct : location in the code
+ * TimePoint 
+
+
+|||
+Detailed Design
+![Logger in details](./assets/detailed_logger_design.png)
+|||
+
+
+### LoggerInitializer interface 
+
+LoggerInitializer specify interface required for Main to init logger but not required for any other SDL components
+
+LoggerInitializer should take and own the third party implementation into `Init` method. 
+ - Init(std::unique_ptr<ThirdPartyLoggerInterface>&& third_party)
+
+
+### ThirdPartyLogger interface
+
+ThirdPartyLogger interface describes interfaces that should be implemented by external logger adapter. 
+
+This interface should be inherited by particular external logger implementation. 
+
+
+## Another logger implementation. 
+
+
+To use another (not log4cxx) logger, you should implement ThirdPartyLoggerInterface class 
+
+```cpp
+AnotherOneLoggerImpl : ThirdPartyLoggerInterface {
+  void Init() override;
+  void DeInit() override;
+  void IsEnabledFor(LogLevel) override;
+  void PushLog(const LogMessage& log_message) override;
+
+  void SomeCustomMethod(parameters);
+}
+```
+
+Then you should create `AnotherOneLoggerImpl` in main and setup it for `LoggerImpl`.
+Note that `Logger::instance` does not own logger instance. `main` function is reponsible for `sdl_logger_instance_` lifecycle.
```

Comment:
Note that `Logger::instance` does not own logger instance. The `main` function is responsible for the `sdl_logger_instance_` life-cycle.

Reply:
Fixed with  
aeadebe